rust/sysdb/src/test_sysdb.rs

```diff
@@ -256,102 +244,107 @@ impl TestSysDb {
+            let mut version_info = CollectionVersionInfo {
+                version: next_version,
+                created_at_secs: if inner.mock_time > 0 {
+                    inner.mock_time as i64
+                } else {
+                    chrono::Utc::now().timestamp()
+                },
+                version_change_reason: VersionChangeReason::DataCompaction as i32,
+                ..Default::default()
+            };
 
-        let mut segment_info = CollectionSegmentInfo::default();
-        let mut flush_compaction_infos = Vec::new();
+            let mut segment_info = CollectionSegmentInfo::default();
+            let mut flush_compaction_infos = Vec::new();
 
-        // Iterate through all segment flush infos
-        for segment_flush_info in segment_flush_info.iter() {
-            let flush_compaction_info: FlushSegmentCompactionInfo = segment_flush_info
-                .try_into()
-                .expect("Failed to convert SegmentFlushInfo");
-            flush_compaction_infos.push(flush_compaction_info);
-        }
+            // Iterate through all segment flush infos
+            for segment_flush_info in segment_flush_info.iter() {
+                let flush_compaction_info: FlushSegmentCompactionInfo = segment_flush_info
+                    .try_into()
+                    .expect("Failed to convert SegmentFlushInfo");
+                flush_compaction_infos.push(flush_compaction_info);
+            }
 
-        segment_info.segment_compaction_info = flush_compaction_infos;
-        version_info.segment_info = Some(segment_info);
+            segment_info.segment_compaction_info = flush_compaction_infos;
+            version_info.segment_info = Some(segment_info);
 
-        // Add new version to history
-        version_history.versions.push(version_info);
-        version_file.version_history = Some(version_history);
+            // Add new version to history
+            version_history.versions.push(version_info);
+            version_file.version_history = Some(version_history);
 
-        tracing::debug!(line = line!(), "version_file: \n{:#?}", version_file);
+            tracing::debug!(line = line!(), "version_file: \n{:#?}", version_file);
 
-        // Update the version file name.
-        let version_file_name = format!(
-            "{}{}/{}",
-            VERSION_FILE_S3_PREFIX, collection_id, next_version
-        );
+            // Update the version file name.
+            let version_file_name = format!(
+                "{}{}/{}",
+                VERSION_FILE_S3_PREFIX, collection_id, next_version
+            );
 
-        inner
-            .collection_to_version_file_name
-            .insert(collection_id, version_file_name.clone());
+            inner
+                .collection_to_version_file_name
+                .insert(collection_id, version_file_name.clone());
+
+            (version_file_name, version_file)
+        };
```

Comment:
whitespace/scope changes to make sure the mutex guard isn't held across an await point